weave

```diff
@@ -584,34 +584,68 @@ dns_args() {
     DNS_ARGS="$DNS_ARG $DNS_SEARCH_ARG"
 }
 
-# Perform operation $1 on local DNS database, for container ID $2
-# with an FQDN obtained from the container's configured hostname and
-# domain, at addresses $3, $4 ...
-tell_dns() {
-    METHOD="$1"
-    CONTAINER_ID="$2"
-    shift 2
-    CONTAINER_FQDN=$(docker inspect --format='{{.Config.Hostname}}.{{.Config.Domainname}}.' $CONTAINER_ID 2>/dev/null) || true
-    tell_dns_fqdn $METHOD $CONTAINER_ID $CONTAINER_FQDN $@
+# Execute $@ only if the weaveDNS container is running
+if_dns_running() {
+    if status=$(docker inspect --format='{{.State.Running}}' $DNS_CONTAINER_NAME 2>/dev/null) || [ "$status" != "true" ] ; then
```

Comment:
This conditional is wrong. The original had a `!` at the beginning and was inverted, i.e. the whole expression evaluated to true when the container *wasn't* running. You didn't apply DeMorgan's laws correctly when negating the expression.